### PR TITLE
Hb6

### PR DESCRIPTION
**Pull request by frostpv**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert ShoppingCart entity"
+                    + shoppingCart + " in to data", e);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]:
```suggestion
                    + shoppingCart + " into the database", e);
```

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't update shopping cart " + shoppingCart
+                    + " in data", e);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't update shopping cart " + shoppingCart, e);
```

File: src/main/java/com/dev/cinema/dao/TicketDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            transaction = session.beginTransaction();
+            session.save(ticket);
+            transaction.commit();
+            return ticket;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert ticket entity"
+                    + ticket + " in to data", e);
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/TicketDaoImpl.java:27]:
```suggestion
                    + ticket + " into the database", e);
```

File: src/main/java/com/dev/cinema/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@Entity
+@Table (name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/model/ShoppingCart.java]: Do we need the EAGER fetch type in here?

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Good job :) Left some minor comments

File: src/main/java/com/dev/cinema/Main.java [hunk 5]
```diff
@@ -55,13 +59,11 @@ public static void main(String[] args) {
...
-        } catch (AuthenticationException e) {
-            System.out.println(e);
-        }
-
+        System.out.println(user);
+        user = userService.findByEmail("frostpv@gmail.com").get();
+        shoppingCartService.addSession(movieSession, user);
+        System.out.println(shoppingCartService.getByUser(user));
+        shoppingCartService.clear(shoppingCartService.getByUser(user));
+        System.out.println(shoppingCartService.getByUser(user));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's maybe save shopping cart in a variable?

File: src/main/java/com/dev/cinema/model/ShoppingCart.java (hunk 4 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/model/ShoppingCart.java]: let's remove eager fetching - in general, this is considered bad practice
and find another way to get tickets along with the shopping cart

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/ShoppingCart.java (hunk 4 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/model/ShoppingCart.java]: fixed

File: src/main/java/com/dev/cinema/Main.java (hunk 5 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/Main.java]: fixed

File: src/main/java/com/dev/cinema/dao/TicketDaoImpl.java (hunk 3 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/TicketDaoImpl.java:27]: fixed

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java (hunk 2 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java (hunk 1 above)
Comment (frostpv) [reply src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]: fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**